tests/beta/test_beta_qnode.py

```diff
@@ -898,3 +898,229 @@ def circuit(x, y):
             assert info["num_trainable_params"] == 4
         else:
             assert info["device_name"] == "default.qubit.autograd"
+
+
+class TestTapeExpansion:
+    """Test that tape expansion within the QNode works correctly"""
+
+    @pytest.mark.parametrize(
+        "diff_method,mode",
+        [("parameter-shift", "backward"), ("adjoint", "forward"), ("adjoint", "backward")],
+    )
+    def test_device_expansion(self, diff_method, mode, mocker):
+        """Test expansion of an unsupported operation on the device"""
+        dev = qml.device("default.qubit", wires=1)
+
+        class UnsupportedOp(qml.operation.Operation):
+            num_wires = 1
+            num_params = 1
+            par_domain = "R"
+
+            def expand(self):
+                with qml.tape.QuantumTape() as tape:
+                    qml.RX(3 * self.data[0], wires=self.wires)
+                return tape
+
+        @qnode(dev, diff_method=diff_method, mode=mode)
+        def circuit(x):
+            UnsupportedOp(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        if diff_method == "adjoint" and mode == "forward":
+            spy = mocker.spy(circuit.device, "execute_and_gradients")
+        else:
+            spy = mocker.spy(circuit.device, "batch_execute")
+
+        x = np.array(0.5)
+        circuit(x)
+
+        tape = spy.call_args[0][0][0]
+        assert len(tape.operations) == 1
+        assert tape.operations[0].name == "RX"
+        assert np.allclose(tape.operations[0].parameters, 3 * x)
+
+    def test_no_gradient_expansion(self, mocker):
+        """Test that an unsupported operation with defined gradient recipe is
+        not expanded for both parameter-shift and finite-differences"""

```

Comment:
I don't see where `finite-differences` is being tested here?

Reply:
```suggestion
        not expanded"""
```
oops! I think I wrote the docstring, and later decided that the logic was unaffected by whatever differentiation method was chosen.